web/src/api/auth/login-callback.ts

```diff
@@ -133,6 +150,98 @@ export const auth0Login = withApiAuthRequired(
         invite_id ? urls.signup({ invite_id }) : urls.signup()
       );
     }
+    let invite: InviteQuery["invite"][number] | null = null;
+
+    if (invite_id) {
+      try {
+        const fetchInviteResult = await FetchInviteSdk(client).Invite({
+          id: invite_id,
+        });
+
+        if (!fetchInviteResult) {
+          throw new Error(`Error while fetching invite: ${invite_id}`);
+        }
+
+        if (fetchInviteResult.invite.length > 0) {
+          invite = fetchInviteResult.invite[0];
+        }
+      } catch (error) {
+        logger.error("Error while fetching invite for FetchInviteSdk.", {
+          error,
+        });
+
+        return res.redirect(
+          307,
+          urls.logout({ login_error: LoginErrorCode.Generic })
+        );
+      }
+
+      if (
+        !invite ||
+        !invite.team_id ||
+        new Date(invite.expires_at) <= new Date() ||
+        invite.email !== auth0User.email
```

Comment:
worth logging this as a separate error as it's quite possible users will run into this